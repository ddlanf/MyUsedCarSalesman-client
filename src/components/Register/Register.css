.register-box{
    margin-top: 12%;
    margin-right: auto;
    margin-left: auto;
    margin-bottom: 10vw;
    width: 85%;
    max-width: 500px;
    text-align: center;
    border: #999090 solid 2px;
    background-color: #EAEDEC;
    border-radius: 10%;
}

.register-box > *{
    font-size: 5vw;
}

.register-header-box{
    text-align: center;
    position: relative;
    margin-bottom: 2%;
}

.register-header{
    display: block;
    font-size: 8vw;
    margin-top: 5%;
    margin-bottom: 3%;
    font-family: 'Merriweather Sans', sans-serif;
}

.register-label, .register-input{
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 1%;
}

.register-label{
    margin-bottom: 1%;
    font-size: 5vw;
}

.register-input{
    margin-bottom: 5%;
    margin-top: 0;
    width: 70%;
    font-size: 4.5vw;
}

.register-label{
    margin-bottom: 2%;
    font-size: 5.5vw;
}
  
.register-submit{
    margin: 2% auto;
    margin-bottom: 5%;
    width: 35%;
    height: 8vw;
    max-height: 55px;
    vertical-align: middle;
    font-size: 4vw;
    border-radius: 10%;
    border: #999090 solid 1px;
    background-color: white;
    cursor: pointer;
}

.register-submit:hover{
    background-color: #999090;
    transition: 0.3s;
}

.register-submit:active{
    border-top: grey solid 2px;
    background-color: #999090;
    padding-top: 0.5%;
}


.register-error{
    color: red;
    font-size: 4vw;
    margin: 0;
    margin-bottom: 5%;
    border: 1px solid #999090;
    background-color: #DFE32E;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
}

.register-login-heading{
    font-size: 5vw;
    width: 90%; 
    margin-top: 0;
    margin-left: auto;
    margin-right: auto;
}

.register-login-link{
    display: block;
    margin-top: 2%;
    color: blue;
}

.register-login-link:hover{
    text-decoration: underline;
}

  
@media screen and (min-width: 1000px) {
  
      .register-box > *{
        font-size: 30px;
      }
  
      .register-header{
        font-size: 55px;
      }

      .register-label, .register-input{
        font-size: 30px;
      }

      .register-error{
        font-size: 30px;
      }

      .register-submit{
          font-size: 30px;
      }  
}