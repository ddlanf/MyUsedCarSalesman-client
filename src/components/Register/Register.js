import React, { Component } from 'react'
import './Register.css'
import AuthApiService from '../../services/auth-api-service'
export default class Register extends Component {

    constructor(props){
        super(props);
        this.state = { 
            first_name: '',
            last_name: '',
            user_name: '',
            password: '',
            email: ''
        }
    }

    registerUser = (ev) =>{
        ev.preventDefault()
        const user = this.state

        let validUser = true

        for(let [key, value] of Object.entries(user)){
            if(value == null || value === ''){
                this.setState({
                    error: `Please enter ${key.replace('_', ' ')}`
                })
                validUser = false
            }
            if(value.trim() !== value){
                this.setState({
                    error: `Inputs cannot contain space`
                })
                validUser = false
            }
        }

        if(validUser){

            AuthApiService.postUser(user)
                .then(message => {
                    AuthApiService.postLogin(user)
                    .then(message => {
                        this.props.history.push('/view-posts')
                        this.props.userLogIn()
                    })
                    .catch(res => {
                        this.setState({ error: res.error })
                    })
                })
                .catch(res => {
                    this.setState({ error: res.error })
                })
        }
    }

    handleInputChange = (event) =>{
        const { name } = event.target
        const { value } = event.target
        
        this.setState({
            [name] : value
        })

    }

    render() {
        return (
            <>
            <div className="register-box">
                <h1 className="register-header">Register</h1>
                <form className="register-form" 
                    onSubmit={this.registerUser}
                    >
                    <label name="first-name" className="register-label">First Name</label>
                    <input 
                        name="first_name" 
                        type="text" 
                        className="register-input"
                        onChange={this.handleInputChange}/>
                    <label name="last-name" className="register-label">Last Name</label>
                    <input 
                        name="last_name" 
                        type="text" 
                        className="register-input"
                        onChange={this.handleInputChange}/>
                    <label name="user-name" className="register-label">User Name</label>
                    <input 
                        name="user_name" 
                        type="text" 
                        className="register-input"
                        onChange={this.handleInputChange}/>
                    <label name="password" className="register-label">Password</label>
                    <input 
                        name="password" 
                        type="password" 
                        className="register-input"
                        onChange={this.handleInputChange}/>
                    <label name="email" className="register-label">Email</label>
                    <input 
                        name="email" 
                        type="text" 
                        className="register-input"
                        onChange={this.handleInputChange}/>
                    <input 
                        type="submit" 
                        className="register-submit"/>
                </form>
                <p className="register-error">{this.state.error}</p>
            </div>
           
            </>
        )
    }
}
