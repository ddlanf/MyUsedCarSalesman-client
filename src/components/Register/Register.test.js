import React from 'react';
import ReactDOM from 'react-dom';
import Register from './Register';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('Register component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <Register
                          isUserLoggedIn={false} 
                          userLogIn={function(){}}
                          userIsAtLoginOrRegister={function(){}}
                          userIsNotAtLoginOrRegister={function(){}}
                          isUserAtLoginOrRegister={false}
                        />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
            <BrowserRouter>
                <Register
                isUserLoggedIn={false} 
                userLogIn={function(){}}
                userIsAtLoginOrRegister={function(){}}
                userIsNotAtLoginOrRegister={function(){}}
                isUserAtLoginOrRegister={false}
                />
            </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});