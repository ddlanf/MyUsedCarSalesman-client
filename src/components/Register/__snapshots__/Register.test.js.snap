// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders the UI as expected 1`] = `
<div
  className="register-box"
>
  <h1
    className="register-header"
  >
    Register
  </h1>
  <form
    className="register-form"
    onSubmit={[Function]}
  >
    <label
      className="register-label"
      name="first-name"
    >
      First Name
    </label>
    
    <input
      className="register-input"
      name="first_name"
      onChange={[Function]}
      type="text"
    />
    <label
      className="register-label"
      name="last-name"
    >
      Last Name
    </label>
    
    <input
      className="register-input"
      name="last_name"
      onChange={[Function]}
      type="text"
    />
    <label
      className="register-label"
      name="user-name"
    >
      Username
    </label>
    
    <input
      className="register-input"
      name="user_name"
      onChange={[Function]}
      type="text"
    />
    <label
      className="register-label"
      name="password"
    >
      Password
    </label>
    
    <input
      className="register-input"
      name="password"
      onChange={[Function]}
      type="password"
    />
    <label
      className="register-label"
      name="email"
    >
      Email
    </label>
    
    <input
      className="register-input"
      name="email"
      onChange={[Function]}
      type="text"
    />
    
    <button
      className="register-submit"
      type="submit"
    >
      Submit
    </button>
  </form>
  <h3
    className="register-login-heading"
  >
    Returning User?
    <span
      className="break"
    >
      <a
        className="register-login-link"
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </span>
  </h3>
</div>
`;
